### DISCUSSION TITLE
Page View Counter

### DISCUSSION THREAD

**Discussion by Rodien** [2sic/2sxc#2417] [Q&A] [answered]
Guys,

Has anyone tried to implement a page view counter for 2sxc?

e.g. a counter for the blog or news app. 

Or have anyone an idea on how we can accomplish this? 

I guess I can make something like this => 
`

var	value = (bool)HttpContext.Current.Session["pageVisited"];
    HttpContext.Current.Session["pageVisited"] = false;
  
  	if(value == false)
    {
       HttpContext.Current.Application ["PageCounter"] = Convert.ToInt32(HttpContext.Current.Application ["PageCounter"]) + 1;
       HttpContext.Current.Session["pageVisited"] = true;
    }
`

And save **PageCounter** in the database. 

But won't that be hard on the database?

This would be a fantastic addition to 2sxc.

**Comment by Rodien** [comment]
Okay, so I found this on stack overflow... 
https://stackoverflow.com/questions/67469687/how-can-we-sort-news-by-top-news-in-2sxc/67488281#67488281


"So there's nothing built in, but since it's all open-source, you could easily add it.

Note that 2sxc data model is optimized for infrequent writing and mass-reading scenarios, so updating data on each view wouldn't be the best way to do this. But you could just put it in a SQL table or something or use another counting/tracking system for that." - iJungleBoy

**Reply by iJungleboy** [reply]
It's funny because I thought you must have been the author on SO ;)

**Comment by iJungleboy** [comment, marked as answer]
Simple counters are often useful, but everybody will have different needs and I don't think it's a priority to add it as a 2sxc feature, also because any developer can quickly brew their own. 

In terms of DB load I would recommend a basic setup where you only update it every 10 or 100 or so counts. 

Or you could simply use an existing web service like https://countapi.xyz/
